### PR TITLE
SWORD1: Add keymapper support

### PR DESCRIPTION
**Pull request by NabeelShabbir**

### CODE REVIEW

File: engines/sword1/metaengine.cpp [hunk 1]
```diff
@@ -103,6 +109,59 @@ Common::Error SwordMetaEngine::createInstance(OSystem *syst, Engine **engine, co
...
+	gameKeyMap->addAction(act);
+
+	act = new Action("QUIT", _("Quit Game"));
+	act->setCustomEngineActionEvent(kActionQuit);
+	act->addDefaultInputMapping("C+q");
+	act->addDefaultInputMapping("JOY_CENTER");
+	gameKeyMap->addAction(act);
+
+	// I18N: Panel is the options screen that is displayed
+	act = new Action("MAINPANEL", _("Main Panel"));
```
Comment (AndywinXp) [inline engines/sword1/metaengine.cpp:152]: Admittedly "Panel" sounds weird, it's just the name used within the source code 🙂 I'm thinking maybe `_("Main Menu")` will work fine here.

File: engines/sword1/sword1.cpp [hunk 2]
```diff
@@ -1098,6 +1101,8 @@ void SwordEngine::pollInput(uint32 delay) { //copied and mutilated from sky.cpp
 	do {
 		while (_eventMan->pollEvent(event)) {
 			switch (event.type) {
+			case Common::EVENT_CUSTOM_ENGINE_ACTION_START:
+				_customType = event.customType;
```
Comment (mgerhardy) [inline engines/sword1/sword1.cpp:1105]: missing `break` here

**Comment by bluegr** [comment]
@NabeelShabbir at a first glance, this looks OK. What else is missing?

**Comment by NabeelShabbir** [comment]
> @NabeelShabbir at a first glance, this looks OK. What else is missing?

Nothing hopefully, I am satisfied with the changes

**Review by OMGPizzaGuy — APPROVED** [review]
Looks good to me

**Comment by bluegr** [comment]
Looks good to me too, nice work!
Squashing (for the merge commit)

**Pull request merged**